sdk/opendp/smartnoise/sql/private_reader.py

```diff
@@ -32,9 +32,17 @@ def __init__(self, metadata, reader, epsilon_per_column=1.0, delta=10E-16,
             :param options: A PrivateReaderOptions with flags that change the behavior of the privacy
                 engine.
         """
-        self.options = options if options is not None else PrivateReaderOptions()
-        self.metadata = metadata
-        self.reader = reader
+        if isinstance(reader, Reader):
+            self.reader = reader
+        else:
+            raise ValueError("Parameter reader must be of type Reader")
+
+        # using string here, because we don't want to import .metadata due to circular reference
+        if "metadata.collection.CollectionMetadata" in str(type(metadata)):
```

Comment:
isinstance(metadata, metadata.collection.CollectionMetadata)

Reply:
isinstance doesn't work here, because you would need to import it, and that creates a circular reference.